### ISSUE TITLE
Use a single bucket for Huly

### ISSUE THREAD

**Issue by NiklasRosenstein** [hcengineering/platform#5713]
### Description of the issue

As far as I can discern, it appears that Huly tries to create a new S3 bucket for each Workspace. I'm encountering this because I have an existing MinIO instance, there doesn't seem to be an option to specify any single bucket and creating a workspace fails with this error:

```
update { createProgress: 20 }
create minio bucket workspace="w-niklas-test-665ccad3-e545e9bc94-ba007d" workspaceId={"name":"w-niklas-test-665ccad3-e545e9bc94-ba007d","productId":""}
failed to create workspace err={"name":"AIe","code":"AccessDenied","message":"Access Denied.","bucketname":"w-niklas-test-665ccad3-e545e9bc94-ba007d","resource":"/w-niklas-test-665ccad3-e545e9bc94-ba007d","requestid":"17D547561B44D5F5","hostid":"dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8","amzRequestid":"17D547561B44D5F5","amzId2":"dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8"} workspaceName="test" email="niklas@rosenstein.email"
Unhandled Rejection at: Promise {
  <rejected> AIe: Access Denied.
      at zVt (/usr/src/app/bundle.js:411:4230)
      at Object.D$n [as parseResponseError] (/usr/src/app/bundle.js:411:4917)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async nt.makeRequestStreamAsync (/usr/src/app/bundle.js:417:2307)
      at async nt.getBucketRegionAsync (/usr/src/app/bundle.js:417:2785)
      at async nt.makeRequestStreamAsync (/usr/src/app/bundle.js:417:1709)
      at async nt.makeRequestAsyncOmit (/usr/src/app/bundle.js:417:789) {
    code: 'AccessDenied',
    bucketname: 'w-niklas-test-665ccad3-e545e9bc94-ba007d',
    resource: '/w-niklas-test-665ccad3-e545e9bc94-ba007d',
    requestid: '17D547561B44D5F5',
    hostid: 'dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8',
    amzRequestid: '17D547561B44D5F5',
    amzId2: 'dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8',
    amzBucketRegion: undefined
  }
} reason: AIe: Access Denied.
    at zVt (/usr/src/app/bundle.js:411:4230)
    at Object.D$n [as parseResponseError] (/usr/src/app/bundle.js:411:4917)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async nt.makeRequestStreamAsync (/usr/src/app/bundle.js:417:2307)
    at async nt.getBucketRegionAsync (/usr/src/app/bundle.js:417:2785)
    at async nt.makeRequestStreamAsync (/usr/src/app/bundle.js:417:1709)
    at async nt.makeRequestAsyncOmit (/usr/src/app/bundle.js:417:789) {
  code: 'AccessDenied',
  bucketname: 'w-niklas-test-665ccad3-e545e9bc94-ba007d',
  resource: '/w-niklas-test-665ccad3-e545e9bc94-ba007d',
  requestid: '17D547561B44D5F5',
  hostid: 'dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8',
  amzRequestid: '17D547561B44D5F5',
  amzId2: 'dd9025bab4ad464b049177c95eb6ebf374d3b3fd1af9251148b658df7ac2e3e8',
  amzBucketRegion: undefined
}                                                                                                                                                   
```

I was hoping to just give it a `huly-data` bucket. Alternatively, is there a way to configure the prefix that is used when Huly creates a new bucket? At the moment that appears to be `w`, though I would like to choose something more distinctive among my other buckets.

### Your environment
* Version of huly: v0.6.230

### Steps to reproduce

Give Huly access credentials to MinIO that can't create a bucket.

**Comment by haiodo** [comment]
for s3 is possible to specify rootBucket variable to pass all workspaces into one bucket.

**Comment by NiklasRosenstein** [comment]
Hi @haiodo , thanks I'm looking into this again right now. How do I set this configuration value? There doesn't seem to be an environment variable that is used for this option.

**Comment by DyllasDek** [comment]
Any movement on that?

**Comment by bichle-mechmaster** [comment]
for anyone facing this problem in the future, and since it's not been documented yet, you can try this solution, this is for minio storage

**TLDR**: Add `rootBucket=<your-bucket>` to the query string

```bash
STORAGE_CONFIG="minio|minio:9000?accessKey=admin&secretKey=secret&rootBucket=<your-bucket>&useSSL=false&region=us-east-1"
```

Explaination:

https://github.com/hcengineering/platform/blob/2d3ebb61cc39a2ece9f11b0e98b70ad79866cebc/server/minio/src/index.ts#L41-L53
https://github.com/hcengineering/platform/blob/2d3ebb61cc39a2ece9f11b0e98b70ad79866cebc/server/minio/src/index.ts#L78-L80
as we can see, the source code already supported putting everything in a single bucket

now we need to find a way to pass the `rootBucket` to the config object

below is the function to parse the config string to object
https://github.com/hcengineering/platform/blob/2d3ebb61cc39a2ece9f11b0e98b70ad79866cebc/server/server-storage/src/starter.ts#L66-L68
extra url query params will be assigned to the config object
so all we need is to add `rootBucket=<your-bucket>` to the storage config string, it will be parsed and get assigned to the config object like

```ts
config["rootBucket"] = "<your-bucket>"
```

you can use this method to provide more parameters to the config object (if it's supported)

**Comment by moonlight8978** [comment]
for anyone facing this problem in the future, and since it's not been documented yet, you can try this solution, this is for minio storage

**TLDR**: Add `rootBucket=<your-bucket>` to the query string

```bash
STORAGE_CONFIG="minio|minio:9000?accessKey=admin&secretKey=secret&rootBucket=<your-bucket>&useSSL=false&region=us-east-1"
```

Explaination:

https://github.com/hcengineering/platform/blob/2d3ebb61cc39a2ece9f11b0e98b70ad79866cebc/server/minio/src/index.ts#L41-L53
https://github.com/hcengineering/platform/blob/2d3ebb61cc39a2ece9f11b0e98b70ad79866cebc/server/minio/src/index.ts#L78-L80
as we can see, the source code already supported putting everything in a single bucket

now we need to find a way to pass the `rootBucket` to the config object

below is the function to parse the config string to object
https://github.com/hcengineering/platform/blob/2d3ebb61cc39a2ece9f11b0e98b70ad79866cebc/server/server-storage/src/starter.ts#L66-L68
extra url query params will be assigned to the config object
so all we need is to add `rootBucket=<your-bucket>` to the storage config string, it will be parsed and get assigned to the config object like

```ts
config["rootBucket"] = "<your-bucket>"
```

you can use this method to provide more parameters to the config object (if it's supported)
